### PR TITLE
[TTAHUB-455] add goals & objectives tab to recipient record

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change
- Add new feature flag migration (I didn't write a down method - not sure how to do that in this case?) Glad to add it to this PR if someone can help me with some direction. Even what to Google.
- Add tab to recipient record page, hidden behind feature flag
- Refactor ```<FeatureFlag>``` component to use the existing context that was floating around instead of insisting that the user object be passed down the component chain

## How to test

Ensure that you can see the tab as an admin or a user with the feature flag. Ensure that non-admins/non-users can't see it.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-455

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [n/a] API Documentation updated
- [n/a] Boundary diagram updated
- [n/a] Logical Data Model updated
- [n/a ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

**Review by AdamAdHocTeam — COMMENTED** [review]
It seems I always have the Goals tab regardless of the feature flag value.

**Comment by thewatermethod** [comment]
> It seems I always have the Goals tab regardless of the feature flag value.

Admins have access to all feature flags, are you an admin?

**Review by AdamAdHocTeam — APPROVED** [review]
Works as expected.

File: frontend/src/components/FeatureFlag.js [hunk 1]
```diff
@@ -16,10 +21,6 @@ export default function FeatureFlag({
 
 FeatureFlag.propTypes = {
   flag: PropTypes.string.isRequired,
-  admin: PropTypes.bool.isRequired,
-  user: PropTypes.shape({
-    flags: PropTypes.arrayOf(PropTypes.string),
-  }).isRequired,
```
Comment (jasalisbury) [inline frontend/src/components/FeatureFlag.js:22]: :rocket:

File: src/migrations/20220111194358-add-goals-objective-flag.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+      DO $$ BEGIN
+        ALTER TYPE "enum_Users_flags" ADD VALUE 'recipient_goals_objectives';
+      EXCEPTION
+        WHEN duplicate_object THEN null;
+      END $$;
+    `);
+  },
+
+  down: async () => {
+    // Is there a way to revert this? Not sure
```
Comment (jasalisbury) [inline src/migrations/20220111194358-add-goals-objective-flag.js:13]: https://www.postgresql.org/docs/current/datatype-enum.html section 8.7.4. Removing existing values from the enum is not supported. To remove the old values we'd have to drop the type and add it again.

**Review by jasalisbury — APPROVED** [review]
:shipit:

**Pull request merged**